tls-client/main.cpp

```diff
@@ -132,6 +133,8 @@ class HelloHTTPS {
         mbedtls_x509_crt_free(&_cacert);
         mbedtls_ssl_free(&_ssl);
         mbedtls_ssl_config_free(&_ssl_conf);
+        _tcpsocket->close();
```

Comment:
Ok, then let's do that: move the code that opens the connection to the constructor!